File: Linked_Lists/go/partition.go
package main

import (
  "types"
  "math/rand"
)


func delete(node *types.Node) {
  node.Data = node.Next.Data
  node.Next = node.Next.Next
}


func performPartition(list types.LinkedList, partition int) types.LinkedList {
  node := list.Head

  for node.Next != nil {
    if node.Data < partition && node != list.Head {
      list.AddToBeginning(node.Data)
      delete(node)
    } else {
      node = node.Next
    }
  }

  if node.Data < partition && node != list.Head {
    list.AddToBeginning(node.Data)
    list.DeleteLast()
  }

  return list
}


func main() {
  var list types.LinkedList
  var length int = 5

  for i := 0; i < length; i++ {
    list.Append(rand.Intn(9))
  }

  types.PrintLinkedList(list)
  list = performPartition(list, 6)
  types.PrintLinkedList(list)
}

File: Linked_Lists/python/partition.py
from LinkedList import  LinkedList, Node
from random import randint


def perform_partition(linked_list: LinkedList, partition: int):
    for node in linked_list:
        if node.data < partition and node != linked_list.head:
            linked_list.insert_at_index(0, node.data)

            if not node.next:
                linked_list.delete_last()
            else:
                delete(node)


def delete(node):
    node.data = node.next.data
    node.next = node.next.next


def main():
    linked_list = LinkedList()
    for i in range(10):
        linked_list.append(randint(0, 9))

    print(linked_list)
    perform_partition(linked_list, 6)
    print(linked_list)


if __name__ == '__main__':
    main()
